src/torrent.cpp

```diff
@@ -5684,12 +5686,17 @@ namespace {
 		{
 			m_file_priority = *p;
 			update_piece_priorities();
+
+			if (m_share_mode)
+			{
+				TORRENT_ASSERT(boost::algorithm::all_of_equal(m_file_priority, 0));
```

Comment:
this assert is not a safe assumption, since `m_share_mode` is updated immediately when entering share mode, it doesn't first wait to make sure all outstanding set-priority jobs complete.

Reply:
Let's drop it too